dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java

```diff
@@ -112,6 +112,7 @@ public EventSearchParams map(EventOperationParams operationParams)
     ProgramStage programStage = validateProgramStage(operationParams.getProgramStageUid());
     OrganisationUnit requestedOrgUnit = validateRequestedOrgUnit(operationParams.getOrgUnitUid());
 
+    validateOrgUnitMode(operationParams, user, program);
     OrganisationUnitSelectionMode orgUnitMode =
```

Comment:
This default assignment of orgUnitMode in case something is null shouldn't be called before validation of the orgUnitMode?

What if the default assignment become `ACCESSIBLE` but the user has no orgUnits in search scope?

In my opinion we could move `OrganisationUnitSelectionMode orgUnitMode = getOrgUnitMode(requestedOrgUnit, operationParams.getOrgUnitMode());` in `RequestParamsMapper`, that is generally the place where we assign default values anyway

Reply:
Nice catch!